### PR TITLE
Add ActivityIndexer

### PR DESCRIPTION
**Pull request by maciekzygmunt**

### CODE REVIEW

File: packages/backend/src/modules/activity2/indexers/ActivityIndexer.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+export class ActivityIndexer extends ManagedChildIndexer {
+  constructor(private readonly $: ActivityIndexerDeps) {
+    const logger = $.logger.tag($.projectId)
+    super({ ...$, logger, name: `activity_indexer_${$.projectId}` })
+  }
+
+  override async update(from: number, to: number): Promise<number> {
+    const batchSize = this.$.batchSize ?? 100
```
Comment (antooni) [inline packages/backend/src/modules/activity2/indexers/ActivityIndexer.ts:24]: batch size should always be defined, this ?? will be used during object instantiation

File: packages/backend/src/modules/activity2/indexers/ActivityIndexer.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    const adjustedTo = from + batchSize < to ? from + batchSize : to
+    const counts = await this.$.txsCountProvider.getTxsCount(from, adjustedTo)
```
Comment (antooni) [inline packages/backend/src/modules/activity2/indexers/ActivityIndexer.ts:27]: return array of ActivityRecord from this method

File: packages/backend/src/modules/activity2/indexers/ActivityIndexer.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    const timestamps = Array.from(counts.keys())
+
+    const currentValues =
```
Comment (antooni) [inline packages/backend/src/modules/activity2/indexers/ActivityIndexer.ts:31]: move to method inside this class, e.g. getDatabaseEntries

Comment (antooni) [reply packages/backend/src/modules/activity2/indexers/ActivityIndexer.ts:31]: and also return the array of ActivityRecord

File: packages/backend/src/modules/activity2/indexers/ActivityIndexer.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        [
+          new UnixTime(Math.min(...timestamps)),
+          new UnixTime(Math.max(...timestamps)),
+        ],
+      )
+    const currentMap = new Map(
+      currentValues.map((v) => [v.timestamp.toNumber(), v.count]),
+    )
+
+    const dataToSave = Array.from(counts).map(([timestamp, count]) => {
```
Comment (antooni) [inline packages/backend/src/modules/activity2/indexers/ActivityIndexer.ts:43]: zip arrays, the only thing to decide is what to do when there is no record in the DB, probably in this case we should create a zero record

**Review by maciekop-l2b — APPROVED** [review]
LGTM

**Pull request merged**